raft/node.go

```diff
@@ -484,5 +503,12 @@ func newReady(r *raft, prevSoftSt *SoftState, prevHardSt pb.HardState) Ready {
 	if r.raftLog.unstable.snapshot != nil {
 		rd.Snapshot = *r.raftLog.unstable.snapshot
 	}
+	if r.readIndex != None {
+		c := make([]byte, len(r.readIndexCtx))
+		copy(c, r.readIndexCtx)
```

Comment:
How about just returning this one time instead of returning this every time we return the ready struct?